### ISSUE TITLE
Segmentation Fault importing

### ISSUE THREAD

**Issue by 13bscsaamjad** [esrf-bliss/Lima#103]
Here at MAX IV, we are trying to build the Lima release 1.8.1 with Basler plugin on Cent OS 7.6 machine installed with Python 2.7.5. The Pylon version is 5.0.5. For building we had to change the file `Lima/common/sip/Debug.sip` at line 123 and 278 to correct the reference to the cpp library.
```
sipCpp = new sipDebParams(a0, getConstStr(cn), getConstStr(ns));
```
is changed to
```
sipCpp = new DebParams(a0, getConstStr(cn), getConstStr(ns))
```
Similar change at line 278. The command we used to build is:
```
./install.sh --git --install-prefix=./install --install-python-prefix=./install/python simulator python basler pytango-server
```
The build went successful but when running the executable from bin with the tango device defined in database, it gave segmentation fault.

The same thing happens if we import Basler or Simulator in iPython.
```
from Lima import Simulator
```

**Comment by sdebionne** [comment]
This sounds like a SIP version issue. We use 4.19.8 for our CI and conda packages. SIP 4.18* is not supported anymore. What version do you use?

**Comment by aureocarneiro** [comment]
@13bscsaamjad at MAX IV we are using only SIP v4.14.6. So this should be the issue @sdebionne . I will try it tomorrow with v4.19.8.

**Comment by 13bscsaamjad** [comment]
yes we were indeed using the older version of sip, i have tried with the newer 4.19.8 version and during compilation i get this error:
```
/build/sip/siplimacorepart9.cpp:51:20: error: ‘std::string sipEvent::getMsgStr()’ marked override, but does not override
      ::std::string getMsgStr() SIP_OVERRIDE;
                    ^
```

**Comment by sdebionne** [comment]
I have never seen this before so not sure what is going on here. But I know that there is an issue with the build system that sometimes prevents the `siplimacorepart???.cpp` to be re-generated when they should. After updating SIP, did you start with a clean build folder?

Nothing should prevent you to rebuild your own libraries but are you aware that we have (prebuilt) conda packages for [Basler](https://anaconda.org/esrf-bcu/lima-camera-basler), and its [tango DS](https://anaconda.org/esrf-bcu/lima-camera-basler-tango)?

**Comment by 13bscsaamjad** [comment]
Yes I started with the clean build. We are not using conda on the production control system machines here at MAX IV.

**Comment by sdebionne** [comment]
I don't want to insist on the Conda thingy, but there are tools to convert from Conda to other packaging system, for instance check [conda-rmps](https://github.com/pelson/conda-rpms) that converts a conda environment into a RPM package.

Now to try to understand your issue, I'll need a bit more info. What version of gcc are you using? It might also help to see the files generated by SIP, could you attach a little `.tar.gz` of your `build\sip` folder (just the cpp files)?

On my local machine, files `Generated by SIP 4.19.8` don't have the `SIP_OVERRIDE` macro and actually I don't see why `override` would be used since `Event` is a base class...

**Comment by 13bscsaamjad** [comment]
I will check the conda conversion and see if that helps.

I am using gcc version 4.8.5. Attached is the build/sip directory.

[build-sip.tar.gz](https://github.com/esrf-bliss/Lima/files/3460287/build-sip.tar.gz)

**Comment by sdebionne** [comment]
I see `Generated by SIP 4.19.18` so we are not using the same SIP version. The reason why we stick with `4.19.8` is that we need pyQt5 (a Qt python binding also built with SIP) to leave in the same python environment and the pyQt5 conda package is built with `4.19.8`. I can't remember the details from the top of my head but some SIP incompatibilities were introduced while only the patch version number was incremented which makes things even more complicated.

Could you try again with SIP `4.19.8` while I check if we can easily fix it for `4.19.18`?

**Comment by 13bscsaamjad** [comment]
Yeah it worked for 4.19.8. Thanks for the help. I really appreciate it.
